pkg/apis/options/providers.go

```diff
@@ -0,0 +1,177 @@
+	LoginURL string `json:"loginURL,omitempty"`
+	// RedeemURL is the token redemption endpoint
+	RedeemURL string `json:"redeemURL,omitempty"`
+	// ProfileURL is the profile access endpoint
+	ProfileURL string `json:"profileURL,omitempty"`
+	// ProtectedResource is the resource that is protected (Azure AD only)
+	ProtectedResource string `json:"resource,omitempty"`
+	// ValidateURL is the access token validation endpoint
+	ValidateURL string `json:"validateURL,omitempty"`
+	// Scope is the OAuth scope specification
+	Scope string `json:"scope,omitempty"`
+	// Prompt is OIDC prompt
+	Prompt string `json:"prompt,omitempty"`
+	// ApprovalPrompt is the OAuth approval_prompt
+	// default is set to 'force'
+	ApprovalPrompt string `json:"approvalPrompt,omitempty"`
+	// AllowedGroups is a list of restrict logins to members of this group
+	AllowedGroups []string `json:"allowedGroups,omitempty"`
+
+	// AcrValues is a string of acr values
+	AcrValues string `json:"acrValues,omitempty"`
+}
+
+type KeycloakOptions struct {
+	KeycloakGroups []string `json:"keycloakGroups,omitempty"`
+}
+
+type AzureOptions struct {
+	// AzureTenant directs to a tenant-specific or common (tenant-independent) endpoint
+	// Default value is 'commmon'
+	AzureTenant string `json:"azureTenant,omitempty"`
+}
+
+type BitbucketOptions struct {
+	// BitbucketTeam sets restrict logins to members of this team
+	BitbucketTeam string `json:"bitbucketTeam,omitempty"`
+	// BitbucketRepository sets restrict logins to user with access to this repository
+	BitbucketRepository string `json:"bitbucketRepository,omitempty"`
+}
+
+type GitHubOptions struct {
+	// GitHubOrg sets restrict logins to members of this organisation
+	GitHubOrg string `json:"githubOrg,omitempty"`
+	// GitHubTeam sets restrict logins to members of this team
+	GitHubTeam string `json:"githubTeam,omitempty"`
+	// GitHubRepo sets restrict logins to collaborators of this repository
+	GitHubRepo string `json:"githubRepo,omitempty"`
+	// GitHubToken is the token to use when verifying repository collaborators
+	// it must have push access to the repository
+	GitHubToken string `json:"githubToken,omitempty"`
+	// GitHubUsers allows users with these usernames to login
+	// even if they do not belong to the specified org and team or collaborators
+	GitHubUsers []string `json:"githubUsers,omitempty"`
+}
+
+type GitLabOptions struct {
+	GitLabGroup    []string `json:"gitlabGroups,omitempty"`
+	GitlabProjects []string `json:"gitlabProjects,omitempty"`
+}
+
+type GoogleOptions struct {
+	// GoogleGroups sets restrict logins to members of this google group
+	GoogleGroups []string `json:"googleGroup,omitempty"`
+	// GoogleAdminEmail is the google admin to impersonate for api calls
+	GoogleAdminEmail string `json:"googleAdminEmail,omitempty"`
+	// GoogleServiceAccountJSON is the path to the service account json credentials
+	GoogleServiceAccountJSON string `json:"googleServiceAccountJson,omitempty"`
+}
```

Comment:
Through all of these, when you put them into the struct, the provider name will be duplicated

Should we remove the provider name prefixes?

Similar for the OIDC prefix on the options below